pkg/innerring/processors/netmap/process_peers.go

```diff
@@ -68,6 +95,35 @@ func (np *Processor) processAddPeer(ev netmapEvent.AddPeer) {
 	}
 }
 
+// Check the new node and allow/reject adding it to netmap.
+func (np *Processor) processAddNode(ev netmapEvent.AddNode) {
```

Comment:
why did you decide to put it here? all the previous flow was about parsing and than handling, it was convenient IMO. now we have a dozen of the old parsers/handlers and the new one that executes script and returns some contract thing as an event, not a known sdk/core structure

Reply:
What's the difference with `processAddPeer`? It's the same thing.

Reply:
i meant, i would call `netmapEvent.Node2Info(&ev.Node)` inside the parser, otherwise, i do not see why we need to parse smth and why we have separated every notification flow into "parser" and "handler". but in the `processAddPeer` it is just bytes so maybe your version is not that different

Reply:
Well, parser still parses, but if you're to take a look at what happens in `processAddPeer` it accepts a byte string from the parser which it then unmarshals. So it takes whatever is the native contract/invocation thing as an input. Same thing here, it takes (parsed) contract structure as an input and does whatever it needs to with it (including conversion into SDK type).